install/preview/entrypoint.sh

```diff
@@ -5,6 +5,10 @@
 
 set -ex
 
+if [ "$1" != "logging" ]; then
+  $0 logging 2>&1 | /prettylog
```

Comment:
If I'm reading this correctly, we're re-launching this script in order to redirect all logs to `/prettylog`, correct?

If we are relaunching the script with the intent described above, we can use `exec` to redirect stdout/stderr, with something like this:

```
exec > >(/prettylog) 2>&1
```

This way we don't need to relaunch the script; we can launch `/prettylog` with [process substitution](https://tldp.org/LDP/abs/html/process-sub.html) and [redirect stderr/stdout with exec](https://tldp.org/LDP/abs/html/x17974.html).

This is another approach to the problem, but isn't necessarily better or worse. If you like the above approach we can adopt it; if what you have is more readable then let's keep it as-is. 😄

Reply:
Following up on this, we're also relying on the `set -e` to exit this script upon error, and we're relying on `/entrypoint.sh` and `prettylog` to exit with a non-zero exit value. When the user hits ctrl-c to kill the container, the forked processes exit non-zero then the behavior of `set -e` in the parent process kicks in and terminates the script.

If the `set -e` is removed for whatever reason, then when the forked processes exit we'll proceed to run this script again, which isn't intended. I think this code will be a little bit more correct to have an `exit` after `$0 logging`; that way changes in exit codes or other behaviors won't accidentally run the script twice.